lib/travis/api/v3/queries/log.rb

```diff
@@ -0,0 +1,7 @@
+module Travis::API::V3
+  class Queries::Log < Query
+    def find(job)
```

Comment:
I think this is where the code that checks for the log in the DB, and then fetches it from S3 if not, will go.